openlibrary/core/community_batch_imports.py

```diff
@@ -0,0 +1,140 @@
+from dataclasses import dataclass
+import json
+import hashlib
+
+from json.decoder import JSONDecodeError
+
+from pydantic import ValidationError
+from pydantic_core import ErrorDetails
+from openlibrary import accounts
+from openlibrary.core.imports import Batch
+from openlibrary.plugins.importapi.import_edition_builder import import_edition_builder
+from openlibrary.catalog.add_book import (
+    IndependentlyPublished,
+    PublicationYearTooOld,
+    PublishedInFutureYear,
+    SourceNeedsISBN,
+    validate_record,
+)
+
+
+def generate_hash(data: bytes, length: int = 20):
+    """
+    Generate a SHA256 hash of data and truncate it to length.
+
+    This is used to help uniquely identify a community_batch_import. Truncating
+    to 20 characters gives about a 50% chance of collision after 1 billion
+    community hash imports. According to ChatGPT, anyway. :)
+    """
+    hash = hashlib.sha256()
+    hash.update(data)
+    return hash.hexdigest()[:length]
+
+
+@dataclass
+class CbiErrorItem:
```

Comment:
I made this change, but because it's doesn't inherit from `Exception` and isn't an error this naming looks a bit misleading to me. What about something like `InvalidImportItem`? I'm trying to avoid using `record` or `rec` and also trying to avoid giving the impression this thing is itself an error.

Reply:
Yeah, I was worried about that as well; but I think this _is_/_could be_ an error; just instead of raising it, we're capturing/aggregating them. It even has a lot of the same fields as an Exception! Effectively something like https://docs.python.org/3/tutorial/errors.html#raising-and-handling-multiple-unrelated-exceptions . So I think this seems less confusing on my end :+1: 

Another option is something like `BatchLineStatus` and have it be something like

```
class BatchLineStatus:
    line_number: int
    error_message: str | None

    @property
    def success():
        return bool(error_message)
```

But I think what's here is good :+1: Up to you.